### PR TITLE
integrated the webview and added a pref to choose between internal an…

### PR DESCRIPTION
**Pull request by lhein**
…d external browser #36 #74

Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,7 +48,22 @@
...
+			"type": "object",
+			"title": "AtlasMap Settings",
+			"properties": {
+				"atlasmap.openInBrowser": {
+					"type": "string",
+					"enum": [
+						"Internal",
+						"External (Default OS Browser)"
+					],					
+					"default": "Internal",
```
Comment (apupier) [inline package.json]: I think the default should be external until the "internal" is good enough

**Comment by lhein** [comment]
> * the tests are not letting the default value when finished. it might have side effect in later development. i think in "after" method, a call to set back to default could fix that.

I don't see the problem. Its a test environment. We set the correct value before each start. 

> * is it possible to use constants in TypeScripts to avoid repeating the Strings "Internal"/"External (Default OS browser)"?

I think I already used constants where possible. What other places do u have in mind ?

>     * I think the readme should be improved to mention the existing settings to choose between the Internal or External browser

Will update. Thanks for the reminder.

> * would be nice to have an improved test (or another one?) for "Internal" case checking that the WebView has loaded correctly.

I will create another issue for adding that test.

File: package.json (hunk 1 above)
Comment (lhein) [reply package.json]: agreed

**Comment by apupier** [comment]
> > * the tests are not letting the default value when finished. it might have side effect in later development. i think in "after" method, a call to set back to default could fix that.
> 
> I don't see the problem. Its a test environment. We set the correct value before each start.

It is set before each start in the existing test.
If tomorrow another developer comes and creates a test which is not supposed to be related to this configuration but relying on the default value: When playing only the test, it will work. When playing all tests in the same launch, it will not work.

This can take time to troubleshoot.

It is a general good practice to keep the environment as default when the test is finished to avoid any side-effect with other tests.

**Comment by apupier** [comment]
> > * is it possible to use constants in TypeScripts to avoid repeating the Strings "Internal"/"External (Default OS browser)"?
> 
> I think I already used constants where possible. What other places do u have in mind ?

I should have been more precise :-)

it is here https://github.com/jboss-fuse/vscode-atlasmap/pull/76/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R59 (but in package.json i'm not sure we can reuse shared constants , it might be a special case for this one)

there is an "Internal" here:
https://github.com/jboss-fuse/vscode-atlasmap/pull/76/files#diff-45327f86d4438556066de133327f4ca2R123

and another "Internal" here: https://github.com/jboss-fuse/vscode-atlasmap/pull/76/files#diff-8978c889b7ce48b413936b0ec31bf20fR20

maybe creating a constant "Internal" which can be referenced in both places would be nice. I imagine an equivalent to an Enum in Java. (don't know what it would be in TypeScript)

**Comment by apupier** [comment]
> > * would be nice to have an improved test (or another one?) for "Internal" case checking that the WebView has loaded correctly.
> 
> I will create another issue for adding that test.

agree
makes sense splitting with another task.

**Comment by lhein** [comment]
created https://github.com/jboss-fuse/vscode-atlasmap/issues/77

**Pull request merged**